application/src/main/kotlin/net/corda/v5/application/messaging/FlowMessaging.kt

```diff
@@ -118,4 +119,69 @@ interface FlowMessaging {
      */
     @Suspendable
     fun initiateFlow(x500Name: MemberX500Name, flowContextPropertiesBuilder: FlowContextPropertiesBuilder): FlowSession
+
+    /**
+     * Suspends until a message has been received for each session in the specified [sessions].
+     *
+     * Consider [receiveAllMap(sessions: Map<FlowSession, Class<out Any>>): Map<FlowSession, Any>] when sessions are
+     * expected to receive different types.
+     *
+     * Remember that when receiving data from other parties the data should not be trusted until it's been thoroughly
+     * verified for consistency and that all expectations are satisfied, as a malicious peer may send you subtly
+     * corrupted data in order to exploit your code.
+     *
+     * @param receiveType type of object to be received for all [sessions].
+     * @param sessions Set of sessions to receive from.
+     * @returns a [List] containing the objects received with the same order of [sessions].
```

Comment:
Sorry, minor nitpick. Is the list ordering defined? The problem is that we ask for a `Set` on input, so I don't think we know anything about the session input ordering?